### ISSUE TITLE
Incomplete .after doc 

### ISSUE THREAD

**Issue by Alexsey** [andrewplummer/Sugar#459]
From this example

```
var g = function (a) {
    console.log('!')
    return a
}.after(3)

console.log(g(1), g(1), g(1), g(1))
```

which outputs

```
!
undefined undefined [ [ 1 ], [ 1 ], [ 1 ] ] undefined
```

we can see two undocumented features of <code>.after</code>:
1. When it actually called, it gets array of arrays of all arguments it was called before
2. Subsequent calls are ignored

I think this should be in the docs

Especially because <code>lodash</code> and <code>underscore</code> <code>.after</code> both behave differently:

```
!
!
undefined undefined 1 1
```

**Comment by andrewplummer** [comment]
OK did a lot of work on this. Lots of changes to behavior and some quick decisions I had to make. I'll explain my thinking but we can discuss.

- executing every N times instead of AFTER N times

This was kind of like an "intended" bug. It was intended but just dumb, given the naming. I've changed the code so that it will now always execute AFTER N times.

- collecting arguments

I think we agreed that there was value being added here, so this is staying the same.

- calling immediately if `0` is passed

This was also intentional but even though I think an argument can be made for the reasoning, in most cases I think it's quite unexpected. The arguments for it are weak at best and the arguments against it are stronger I think (namely, 1. the name is `after` so having the callback be executed before the "after" function is even constructed, let alone executed, is very unexpected... 2. this way it can be more aligned with other libs). Bottom line doing this was just being too clever.

- calling more than once vs. single time

This came up in the course of our discussion, although mostly I think because it wasn't clear what the method was doing at first. However I think that the naming `after` makes it ambiguous to whether it should fire multiple times or not, so I created a flag here and made it multiple by default (so that the default is in line with underscore/lodash). Having the ability to do both provides value I think.

- casting

I had a very deep look at this. You were mentioning that casting `false` to a number would be bad, and at the time I agreed, so I even went as far as tightening down other methods to throw errors when unexpected casts were happening. However, this eventually brought me to `String#repeat`, which is part of the ES6 spec, and is a method that Sugar polyfills. Very much like `after`, it accepts only a positive integer. Negative numbers are meaningless, and things like `Infinity` could lock the browser, so instead it throws a RangeError. However, it does accept things like `undefined`, `NaN`, `true`, and `false` and appears to cast them to numbers (`true` will result in repeating once, for example).

Anyway, I was able to do some good refactoring by bringing the casting code of `after` and the `repeat` polyfill together, and given that it's part of the ES6 spec it seems like maybe this is a good standard to follow. So at the moment:

_ after() == 0
_ after(null) == 0
_ after('string') == 0
_ after(NaN) == 0
_ after(false) == 0
_ after(true) == 1 (effectively same as 0 though)
_ after(1.5) == 1


There are other example of the ES6 spec doing some casting that may be less than intuitive, but all things considered, I would prefer to stay close to that if and when it makes sense. You might disagree though... let me know what you think?

**Comment by andrewplummer** [comment]
Also should mention that I went through the docs and made sure they're a lot clearer too (this will be updated when the next version is released).

**Comment by Alexsey** [comment]
Absolutely agree with you about all but calling multiple times vs. single time - there is more expressive and readable whey to achieve single time execution than additional argument - just to call `.once` before `.after`.

**Comment by andrewplummer** [comment]
Sorry, but I'm not sure what you mean. `.once` is an entirely different method that executes only once. What I meant above is the question of whether the function should execute "execute only once after `n` calls" or "always execute after `n` calls", which is quite different.

**Comment by andrewplummer** [comment]
Ah wait I think I see what you mean now:

```
someFunction.after(5).once();
```

Is that right? That's actually a very good point. This would get rid of the need for that argument as well! Only one question:

```
someFunction.once().after(5);
```

Should this also work and would it be confusing how to use?

**Comment by Alexsey** [comment]
well, actually
`someFunction.after(5).once();`
will not work, because it will allow to execute function created by `.after` only once, not `someFunction` itselt. Only
`someFunction.once().after(5);'
will work. I think this example would be nice to have in docs :)

**Comment by andrewplummer** [comment]
Ok yes you're right of course. Actually looking at it that reads in normal English too, so that's an added bonus. Definitely will add to the docs!

**Comment by andrewplummer** [comment]
OK I added a test ensuring that `Function#once` will play nicely with `after` in this way and removed the 2nd argument. Definitely agree that this is better!

**Comment by Alexsey** [comment]
Cool, nice to see it :+1:
